tests/js/client/communication/test-chaos-cluster.js

```diff
@@ -0,0 +1,304 @@
+            } else if (d >= 0.8) {
+              let o = opts();
+              console.warn("RUNNING AQL REMOVE. OPTIONS: " + JSON.stringify(o));
+              query.call(ctx, "FOR doc IN " + c.name() + " LIMIT 138 REMOVE doc IN " + c.name(), {}, o);
+            } else if (d >= 0.7) {
+              let o = opts();
+              console.warn("RUNNING AQL REPLACE. OPTIONS: " + JSON.stringify(o));
+              query.call(ctx, "FOR doc IN " + c.name() + " LIMIT 1338 REPLACE doc WITH { pfihg: 434, fjgjg: 555 } IN " + c.name(), {}, o);
+            } else if (d >= 0.25) {
+              let o = opts();
+              let d = docs();
+              console.warn("RUNNING INSERT WITH " + d.length + " DOCS. OPTIONS: " + JSON.stringify(o));
+              c.insert(d, o);
+            } else {
+              let d = docs();
+              console.warn("RUNNING REMOVE WITH " + d.length + " DOCS");
+              c.remove(d);
+            }
+          } catch (err) {}
+        }
+        
+        if (trx) {
+          if (Math.random() < 0.2) {
+            trx.abort();
+          } else {
+            trx.commit();
+          }
+        }
+      };
+
+      testOpts.collection = cn;
+      code = `(${code.toString()})(${JSON.stringify(testOpts)});`;
+      
+      let tests = [];
+      for (let i = 0; i < 3; ++i) {
+        tests.push(["p" + i, code]);
+      }
+
+      // run the suite for 5 minutes
+      runParallelArangoshTests(tests, 5 * 60, cn);
+
+      print("Finished load test - waiting for cluster to get in sync before checking consistency.");
+      clearAllFailurePoints();
+      waitForShardsInSync(cn);
+      checkCollectionConsistency(cn);
+    }
+  };
+}
+
+function BuildChaosSuite(opts, suffix) {
+  let suite = {};
+  deriveTestSuite(BaseChaosSuite(opts), suite, suffix);
+  return suite;
+}
+
+const params = ["IntermediateCommits", "FailurePoints", "Truncate", "VaryingOverwriteMode", "StreamingTransactions"];
+
+function addSuite(paramValues) {
+  let suffix = "";
+  let opts = {};
+  for (let j = 0; j < params.length; ++j) {
+    suffix += paramValues[j] ? "_with_" : "_no_";
+    suffix += params[j];
+    opts["with" + params[j]] = paramValues[j];
+  }  
+  let func = function() { return BuildChaosSuite(opts, suffix); };
+  // define the function name as it shows up as suiteName
+  Object.defineProperty(func, 'name', {value: "ChaosSuite" + suffix, writable: false});
+
+  jsunity.run(func);
+}
+/*
+for (let i = 0; i < (1 << params.length); ++i) {
+  let paramValues = [];
+  for (let j = params.length - 1; j >= 0; --j) {
+    paramValues.push(Boolean(i & (1 << j)));
+  }
+  addSuite(paramValues);
+}
+*/
+
+// ATM we only create a single suite with all options except Truncate, because there are still known issues.
+// Later we probably want to have all possible combinations, at least for the nightly builds.
+addSuite([true, true, false, true, false]);
```

Comment:
Can this be resolved now, i.e. add all suites here or in another `nightly` file?

Reply:
I am going to add the other suites in a separate PR.